api/src/main/java/org/apache/iceberg/Accessors.java

```diff
@@ -187,6 +206,47 @@ public String toString() {
     }
   }
 
+  private static class DateAccessor extends PositionAccessor {
+    private DateAccessor(int pos, Type type) {
+      super(pos, type);
+    }
+
+    @Override
+    public Object get(StructLike record) {
+      LocalDate value = record.get(position(), LocalDate.class);
+      return (int) ChronoUnit.DAYS.between(EPOCH_DAY, value);
```

Comment:
By moving these into `iceberg-data`, we can reuse the util class `DateTimeUtil`.

Reply:
Does it make sense to have a separate PR to add those common functions into DateTimeUtil?